specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1002 @@
+          "BenefitRecommendationsBillingAccountList": {
+            "$ref": "./examples/BenefitRecommendationsByBillingAccount.json"
+          }
+        },
+        "operationId": "BenefitRecommendations_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/benefitRecommendationsFilterParameter"
+          },
+          {
+            "name": "$orderby",
+            "in": "query",
+            "description": "May be used to order the recommendations by: properties/armSkuName. This implicitly filters the results by the corresponding benefit type.",
+            "type": "string",
+            "x-ms-parameter-location": "method"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "description": "May be used to expand the properties by: properties/usage, properties/allRecommendationDetails",
+            "type": "string",
+            "x-ms-parameter-location": "method"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingScopeBenefitRecommendationParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/benefitRecommendationsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{billingScope}/providers/Microsoft.CostManagement/benefitRecommendations/{recommendationId}": {
+      "get": {
+        "tags": [
+          "BenefitRecommendations"
+        ],
+        "description": "Details of a benefit plan recommendation for what-if analysis.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/CostManagement/"
+        },
+        "x-ms-examples": {
+          "BenefitRecommendationById": {
+            "$ref": "./examples/BenefitRecommendationsByRecommendationId.json"
+          }
+        },
+        "operationId": "BenefitRecommendation_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingScopeBenefitRecommendationParameter"
+          },
+          {
+            "name": "recommendationId",
+            "in": "path",
+            "description": "The ID of the recommendation.",
+            "required": true,
+            "type": "string",
+            "x-ms-parameter-location": "method",
+            "x-ms-skip-url-encoding": true
```

Comment:
what are valid formats/regex patterns/lengths?

Reply:
Recommendation id is generated by API's upstream service. There are number of fields considered while generating this id and logic may change anytime. We don't want to reveal any more information other than its a string to the customer. Customers call list API first to get this id and then make a call to GET API if they want to.

Reply:
So... it is actually format: "arm-id" or format "uri"?

Reply:
Recommendation id here corresponds to the parameter in the request path
/{billingScope}/providers/Microsoft.CostManagement/benefitRecommendations/**_{recommendationId}_**

If we say the format of recommendationId parameter in the request path as 'arm-id' or 'uri',
The request URL will look like this which is not correct. /{billingScope}/providers/Microsoft.CostManagement/benefitRecommendations/**_[{billingScope}/providers/Microsoft.CostManagement/benefitRecommendations}/id]_**

Whereas the correct URL is like below and the last part of the URL is the recommendation id. 
billingAccount/123456/providers/Microsoft.CostManagement/benefitRecommendations/**_0000000000abcdedh00000000000000000_**

But the suggestion applies for the response property format.

Reply:
Thanks for clarifying.